views/podcasts/DisplaySection/PodcastList.tsx

```diff
@@ -4,14 +4,25 @@ import PodcastEpisodeCard from "./PodcastEpisodeCard"
 
 interface PodcastListProps {
   episodes: PodcastEpisode[]
+  currentPage: number
+  itemsPerPage: number
 }
 
-const PodcastList: React.FC<PodcastListProps> = ({ episodes }) => (
-  <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-2 lg:grid-cols-2 gap-x-16 gap-y-24 m-24 ">
-    {episodes.map((episode, index) => (
-      <PodcastEpisodeCard key={index} episode={episode} />
-    ))}
-  </div>
-)
+const PodcastList: React.FC<PodcastListProps> = ({
+  episodes,
+  currentPage,
+  itemsPerPage,
+}) => {
+  const startIndex = (currentPage - 1) * itemsPerPage
+  const endIndex = startIndex + itemsPerPage
+  const episodesToDisplay = episodes.slice(startIndex, endIndex)
+  return (
+    <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-2 lg:grid-cols-2 gap-x-16 gap-y-24 m-24 ">
+      {episodesToDisplay.map((episode, index) => (
+        <PodcastEpisodeCard key={index} episode={episode} />
```

Comment:
Thing of preference but you can do `<PodcastEpisodeCard key={index} {...episode} />` and refactor the `PodcastEpisodeCard` to take deconstructed episode.
Also, I recommend to use episode.id or something similar instead of index.
https://dev.to/shiv1998/why-not-to-use-index-as-key-in-react-lists-practical-example-3e66

In this case, since there are no dynamic changes to the rendered list, it will probably be non-issue BUT
it is a bad practice that can cause hard-to-find bugs and performance issues